woocommerce/woocommerce-ios
# [WOO POS] Update Networking and Yosemite

## Description


Started separating https://github.com/woocommerce/woocommerce-ios/pull/13029 into smaller PRs

- Add createPointOfSaleOrder and updatePointOfSaleOrder in Networking
- Add Yosemite changed and related changes needed for order calculation and sync
- Remaining UI and other changes will be done in a separate PR, both this and a separate PR(s) are targeting **feat/12763-pos-order-totals-calculation-presentation** branch so we do not merge all of this to **trunk** immediately 

## Testing Steps - New order in app
- Open Orders
- Create new order with few products and enter order details
- Check that status of the newly created order is "pending payment"
- Tap "Edit" to edit the order
- Remove one of the products and add a new product
- Edit other fields in the order as well
- Tap "Done" to finish editing the order
- Check that the order is edited correctly and shows the changes
- Go to Collect Payment and make sure that the payment is successful and order updated accordingly

## Testing Steps - Old order in app
- Open Orders
- Select one of the older "pending payment" orders
- Tap "Edit" to edit the order
- Remove one of the products and add a new product
- - Edit other fields in the order as well
- Tap "Done" to finish editing the order
- Check that the order is edited correctly and shows the changes

## Testing Steps POS

- Enter POS
- Add few products to the cart
- Tap "Checkout"
- Check that the same products appear on the left side and totals amount are calculated correctly (for now still with fixed tax of 10%, in next PRs it will show the synced order data)

---
